packages/p2p/src/components/buy-sell/buy-sell-modal.jsx

```diff
@@ -161,97 +222,102 @@ const BuySellModal = ({ table_type, selected_ad, should_show_popup, setShouldSho
-                portalId={general_store.props.modal_root_id}
-                toggleModal={onCancel}
-            >
-                {/* Parent height - Modal.Header height - Modal.Footer height */}
-                <ThemedScrollbars height={table_type === buy_sell.BUY ? '100%' : 'calc(100% - 5.8rem - 7.4rem)'}>
-                    <Modal.Body className='buy-sell__modal--layout'>
-                        {table_type === buy_sell.SELL && is_account_balance_low && <LowBalanceMessage />}
-                        {!!error_message && <BuySellFormError />}
-                        {my_profile_store.should_show_add_payment_method_form ? (
-                            <AddPaymentMethodForm formik_ref={formik_ref} should_show_separated_footer />
-                        ) : (
-                            <Form
-                                advert={selected_ad}
-                                handleClose={onCancel}
-                                handleConfirm={onConfirmClick}
-                                setIsSubmitDisabled={setIsSubmitDisabled}
-                                setErrorMessage={setErrorMessage}
-                                setSubmitForm={setSubmitForm}
-                            />
-                        )}
-                    </Modal.Body>
-                </ThemedScrollbars>
-                {!my_profile_store.should_show_add_payment_method_form && (
-                    <Modal.Footer has_separator>
-                        {my_profile_store.should_show_add_payment_method_form ? null : (
+    return (
+        <React.Fragment>
+            <MarketRateChangeErrorModal
+                is_open={show_market_rate_change_error_modal}
+                closeModal={() => setShowMarketRateChangeErrorModal(false)}
+                setShouldShowPopup={setShouldShowPopup}
+            />
+            <MobileWrapper>
+                <MobileFullPageModal
+                    body_className='buy-sell__modal-body'
+                    className='buy-sell__modal'
+                    height_offset='80px'
+                    is_flex
+                    is_modal_open={should_show_popup}
+                    page_header_className='buy-sell__modal-header'
+                    page_header_text={generateModalTitle(formik_ref, my_profile_store, table_type, selected_ad)}
+                    pageHeaderReturnFn={onCancel}
+                    page_footer_parent={my_profile_store.should_show_add_payment_method_form ? '' : page_footer_parent}
+                    renderPageFooterChildren={() =>
+                        !my_profile_store.should_show_add_payment_method_form && (
                             <BuySellModalFooter
                                 is_submit_disabled={is_submit_disabled}
                                 onCancel={onCancel}
-                                onSubmit={submitForm.current}
+                                onSubmit={!has_rate_changed_recently ? submitForm.current : onSubmitWhenRateChanged}
```

Comment:
Can we change this since we're using a ternary anyway. 

```suggestion
                                onSubmit={has_rate_changed_recently ? onSubmitWhenRateChanged : submitForm.current}
```

Reply:
Oh yaa its much more readable that way, I'll do that asap